components/process.rst

```diff
@@ -352,6 +352,35 @@ The input of a process can also be defined using `PHP streams`_::
     // will echo: 'foobar'
     echo $process->getOutput();
 
+Using TTY and PTY Modes
+-----------------------
+
+All examples above show that your program has control over the input of a
+process (using ``setInput()``) and the output from that process (using
+``getOutput()``). The Process component has two special modes that tweak
+the relationship between your program and the process: teletype (tty) and
+pseudo-teletype (pty).
+
+In TTY mode, you connect the input and output of the process to the input
+and output of your program. This allows for instance to open an editor like
+Vim or Nano as a process. You enable TTY mode by calling
+:method:`Symfony\\Component\\Process\\Process::setTty`::
+
+    $process = new Process(['vim']);
+    $process->setTty(true);
+    $process->run();
+
+    // As the output is connected to the terminal, it is no longer possible
+    // to read or modify the output from the process!
+    dump($process->getOutput()); // null
+
+In PTY mode, your program behaves as a terminal for the process instead of
```

Comment:
`TTY` mode is very clear to me thanks to your previous explanation and example ... but PTY is not the same case. I don't fully understand why/when I should use PTY instead of the default Symfony behavior (which is no TTY and no PTY). Thanks!

Reply:
Yeah, the big issue is that I can't come up with a use-case for PTY. Even more, I've searched github for `setPty(true)` and didn't find anything useful, nor did googling for articles about PHP's own pty support.

So what do you think about significantly shortening this paragraph instead? I think PTY will only be used by process experts, we mainly need to document it to tell them "we support those too". So mentioning `setPty()` somewhere is probably enough (?)

Reply:
After reading some resources about this, I liked this simplification that I read in a StackOverflow answer:

* TTY is a real console, like the one you use in your OS
* PTY is a fake console, like the one you use when doing SSH into another machine

Can we run `ssh` via Symfony's Process when using PTY?